### DISCUSSION TITLE
Help in creating conditional source !

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#1899] [Q&A] [answered]
If I create a conditional source with this code -

```
input = if !toggle then
        input.harbor("test", port=9000, password="hackme")
      else
        input.http(poll_delay=10.,max_buffer=!input_buffer,"#{stream_url}")
      end
```

I get this error
```
Error 5: this value has type
  (...) -> source(?A)
  .{
    time : () -> float,
    shutdown : () -> unit,
    fallible : bool,
    skip : () -> unit,
    seek : (float) -> float,
    is_active : () -> bool,
    is_up : () -> bool,
    log : 
    {level : (() -> int?).{set : ((int) -> unit)}
    },
    self_sync : () -> bool,
    duration : () -> float,
    elapsed : () -> float,
    remaining : () -> float,
    on_track : ((([string * string]) -> unit)) -> unit,
    on_leave : ((() -> unit)) -> unit,
    on_shutdown : ((() -> unit)) -> unit,
    on_metadata : ((([string * string]) -> unit)) -> unit,
    is_ready : () -> bool,
    id : () -> string,
    buffer_length : () -> float,
    status : () -> string,
    connected_client : () -> string?,
    stop : (() -> unit)}
but it should be a subtype of
  (...) -> _.{is_started : _}?
```
at the first line of this code.

If I use the same input either http or harbor in both if and else outcomes all is well, the script runs. ?

**Comment by smimram** [comment, marked as answer]
I would convert those as `sources` explicitely, something like:
```
input = if !toggle then
        (input.harbor("test", port=9000, password="hackme") : source)
      else
        (input.http(poll_delay=10.,max_buffer=!input_buffer,"#{stream_url}") : source)
      end
```
Also it should work better with current Liquidsoap from git.

**Comment by Russsgithub** [comment]
Cool, Thanks Sam, I'll give it ago.

On 6 Oct 2021 20:57, Samuel Mimram ***@***.***> wrote:

I would convert those as sources explicitely, something like:

input = if !toggle then
        (input.harbor("test", port=9000, password="hackme") : source)
      else
        (input.http(poll_delay=10.,max_buffer=!input_buffer,"#{stream_url}") : source)
      end


Also it should work better with current Liquidsoap from git.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/savonet/liquidsoap/discussions/1899#discussioncomment-1438638>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AB627QNBZLHZEYBEAESTVULUFSSZNANCNFSM5D27BJKA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
